pkg/service/rtcservice.go

```diff
@@ -236,27 +229,13 @@ func (s *RTCService) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	// give it a few attempts to start session
 	var cr connectionResult
 	var initialResponse *livekit.SignalResponse
-	for attempt := 0; attempt < startConnectionMaxTries && r.Context().Err() == nil; attempt++ {
```

Comment:
if the context is already canceled when we get here the loop condition is never met and we continue without setting err... this causes a nil pointer exception when we try to access the signal channel